internal/impl/messaging_service_impl.go

```diff
@@ -157,6 +158,46 @@ func (service *messagingServiceImpl) Connect() (ret error) {
 	return nil
 }
 
+func (service *messagingServiceImpl) UpdateProperty(property config.ServiceProperty, value interface{}) (ret error) {
+        // Verify that the value is of type string. This is necessary since the only values that are acceptable for the
+        // properties that are currently supported for modification are strings. We take an interface type as the parameter
+        // for the sake of forwards compatibility.
+        _, valueIsString := value.(string)
+        if !valueIsString {
```

Comment:
This check seems a bit strange to me. It happens that all of our properties currently use only strings. Is there a way for us to instead use the existing property type mappings and validate based on those given a certain property? See the message service property mappings at https://github.com/SolaceDev/pubsubplus-go-client/blob/main/internal/impl/messaging_service_property_mapping.go

This way, if we ever add a new property to the list of properties allowed, we don't have to do any additional work to handle new property types.

Upon reading the service property mapping, we apparently do not impose any type restrictions and simply convert types. Maybe we can remove this check entirely and rely on the converter?

Reply:
I don't think the converter checks the type of the value for the property, or at least does not do it in a way that is accessible to the rest of the API. I think we should have the check done here so that we can reject the value as soon as possible in the operation. This isn't critical for now, but it's not currently useless and it would be very useful to have if in the future the property update became more expensive, like if there were additional checks, or some kind of blocking from CCSMP. I don't know why these would ever exist, but it would be nice to have the infrastructure built in beforehand so that we didn't have to do as much heavy lifting. The map of property to value type could also be used in the converter to validate property values if that is something that you believe the Go API would benefit from.

Reply:
I've removed the type check on the value